crates/walrus-service/src/node.rs

```diff
@@ -1646,49 +1660,82 @@ impl StorageNode {
             .send(Some(event.epoch))?;
 
         // Perform database cleanup operations.
-        // TODO(WAL-1040): Move this to a background task.
-        self.perform_db_cleanup(event.epoch).await?;
+        // Try to get the epoch start time from the contract service. If the epoch state is not
+        // available (e.g., in tests), use the current time as the epoch start.
+        let epoch_start = self
+            .inner
+            .contract_service
+            .get_epoch_and_state()
+            .await
+            .ok()
+            .and_then(|(current_epoch, state)| {
+                if current_epoch == event.epoch {
+                    state.start_of_current_epoch()
+                } else {
+                    None
+                }
+            })
+            .unwrap_or_else(Utc::now);
+        if let Err(error) = self
+            .garbage_collector
+            .start_garbage_collection_task(event.epoch, epoch_start)
+            .await
+        {
+            tracing::error!(
+                ?error,
+                epoch = event.epoch,
+                "failed to start garbage-collection task"
+            );
+        }
```

Comment:
question: IIUC, this is the main change in this PR, in terms of functionality?

Reply:
Yes. Plus the restart and the random start time.